lib/kubernetes-deploy/kubernetes_resource/memcached.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class Memcached < KubernetesResource
+    TIMEOUT = 5.minutes
+    SECRET_NAME = "memcached-url"
+
+    def sync
+      _, _err, st = kubectl.run("get", type, @name)
+      @found = st.success?
+      @deployment_exists = memcached_deployment_exists?
+      @service_exists = memcached_service_exists?
+      @secret_exists = memcached_secret_exists?
+
+      @status = if @deployment_exists && @service_exists && @secret_exists
+        "Provisioned"
+      else
+        "Unknown"
+      end
+    end
+
+    def deploy_succeeded?
+      @deployment_exists && @service_exists && @secret_exists
+    end
+
+    def deploy_failed?
+      false
+    end
+
+    def exists?
+      @found
+    end
+
+    private
+
+    def memcached_deployment_exists?
+      deployments, _err, st = kubectl.run("get", "deployments", "-l name=#{@name}", "-o=json")
```

Comment:
Swapped to using a label selector to verify the deployment exists instead. This prevents us from being forced to name the `Service` and `Deployment` exactly what the template provides or have knowledge we shouldn't about how things are named.

@stefanmb

Reply:
I think that makes sense! Could I ask you to add this to the refactor list for `redis` and `cloudsql` as well? Outside this PR, of course.

Reply:
I will add that to the issue but we will need to make additional changes to make sure the `Redis` and  `Cloudsql` TPR controllers actually add this label. The `Memcached` controller **does** add this label.